Doc/library/math.rst

```diff
@@ -166,6 +166,20 @@ Number-theoretic and representation functions
    Return ``True`` if *x* is a NaN (not a number), and ``False`` otherwise.
 
 
+.. function:: isqrt(n)
+
+   Return the "integer square root" of the integer *n*. This is the integer
+   part of the exact square root of *n*.
+
+   For some applications, it is convenient to have the ceiling of the square
+   root of *n* (the smallest integer ``a`` such that ``n <= a * a``) rather
+   than the floor (the largest integer ``a`` such that ``a * a <= n``). For a
```

Comment:
I afraid that it may be confusing that the actual definition of the function (the largest integer `a` such that `a * a <= n`) goes after the definition of the alternate function. I suggest to move it to the previous paragraph and keep this paragraph for ceiling only.

Reply:
Agreed. I'll move this.